encoding2.c

```diff
@@ -393,7 +393,7 @@ void method_getArgumentType(Method method,
                             size_t dst_len)
 {
 	if (NULL == method) { return; }
-	const char *types = findParameterStart(method->types, index);
+	const char *types = findParameterStart(method->types, index + 1);
```

Comment:
Rather than having the +1 in both places, it would probably be cleaner to fix `findParaneterStart` to start in the right place.